### ISSUE TITLE
Device Tree Modification

### ISSUE THREAD

**Issue by fabian727** [zhemao/zhemao.github.com#23]
Hi,

I read your 'tutorial' Exploring the Arrow Sockit. Great text. Would helped me if I had found it sooner.
But I tried to much myself. Same mistake as you did.
But I am still in learning, so here i am.

I have a cyclone V development board and wanted to modify it with linux (yocto angstrom) and two ethernet busses.

Linux is booting and working as should be.
My problems are the ethernet busses, which aren't recognized by the linux. The device tree I am using is from yocto and so unmodified.
How can I modify it to find the ethernet periphery?
I have the device tree of the yocto project and the device tree of the Quartus v14.1 (I can create any other version device tree too).
But the design of the files are different...

Do you have a tutorial how to modify the dts / dtb of yocto, or how to parse them?

Thanks ahead,
Fabian

**Comment by zhemao** [comment]
You usually don't modify the device tree directly. Instead, you should change the kernel configuration to include the drivers for your new devices. How are your ethernet buses connected to the board?

**Comment by fabian727** [comment]
thanks for your reply,

my Linux is running on the two ARM kernels.

The first ethernet is enabled in the hard processor system.

The other two ethernet systems I created like in this tutorial: http://www.rocketboards.org/foswiki/Projects/AlteraSoCTripleSpeedEthernetDesignExample

Linux doesn't find any ethernet system.
Not the one enabled in the hps nor the created ones in the fpga.

In the kernel configuration I changed nothing and the ethernet driver were as standard enabled.
Also I have a CAN-device on the system which I can see and work with. The internal "network"-interface lo is enabled, too.
